orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java

```diff
@@ -108,55 +96,16 @@ public MutableStatusRegistry lockApplicationInstance_forCurrentThreadOnly(Applic
     MutableStatusRegistry lockApplicationInstance_forCurrentThreadOnly(
             ApplicationInstanceReference applicationInstanceReference,
             long timeoutSeconds) {
-        Thread currentThread = Thread.currentThread();
-
-        //Due to limitations in SessionFailRetryLoop.
-        assertThreadDoesNotHoldLock(currentThread,"Can't lock " + applicationInstanceReference);
+        String lock2Path = applicationInstanceLock2Path(applicationInstanceReference);
+        Lock lock = new Lock(lock2Path, curator);
+        lock.acquire(Duration.ofSeconds(timeoutSeconds));
 
         try {
-            SessionFailRetryLoop sessionFailRetryLoop = 
-                    curator.framework().getZookeeperClient().newSessionFailRetryLoop(Mode.FAIL);
-            sessionFailRetryLoop.start();
-            try {
-                String lockPath = applicationInstanceLockPath(applicationInstanceReference);
-                InterProcessSemaphoreMutex mutex = acquireMutexOrThrow(timeoutSeconds, TimeUnit.SECONDS, lockPath);
-
-                // TODO: Once rolled out, make this the only lock mechanism
-                Lock lock2;
-                try {
-                    String lock2Path = applicationInstanceLock2Path(applicationInstanceReference);
-                    lock2 = new Lock(lock2Path, curator);
-                    lock2.acquire(Duration.ofSeconds(timeoutSeconds));
-                } catch (Throwable t) {
-                    mutex.release();
-                    throw t;
-                }
-
-                synchronized (threadsHoldingLock) {
-                    threadsHoldingLock.put(currentThread, applicationInstanceReference);
-                }
-
-                return new ZkMutableStatusRegistry(
-                        lock2,
-                        mutex,
-                        sessionFailRetryLoop,
-                        applicationInstanceReference,
-                        currentThread);
-            } catch (Throwable t) {
-                sessionFailRetryLoop.close();
-                throw t;
-            }
-        } catch (Exception e) {
-            //TODO: IOException with explanation
-            throw new RuntimeException(e);
-        }
-    }
-
-    private void assertThreadDoesNotHoldLock(Thread currentThread, String message) {
-        synchronized (threadsHoldingLock) {
-            if (threadsHoldingLock.containsKey(currentThread)) {
-                throw new AssertionError(message + ", already have a lock on " + threadsHoldingLock.get(currentThread));
-            }
+            return new ZkMutableStatusRegistry(lock, applicationInstanceReference);
+        } catch (Throwable t) {
```

Comment:
Use try-finally instead.

Reply:
If an exception is NOT thrown I should not close it.

Reply:
Ofc, my mistake 👍